src/Ansatz/Chain.jl

```diff
@@ -113,14 +113,13 @@ function Chain(::Operator, boundary::Open, arrays::Vector{<:AbstractArray})
 end
 
 rightsite(tn::Chain, site::Site) = rightsite(boundary(tn), tn, site)
-rightsite(::Open, tn::Chain, site::Site) = Site(site.id + 1)
+rightsite(::Union{Open, Periodic}, tn::Chain, site::Site) = Site(site.id + 1)
 
 leftsite(tn::Chain, site::Site) = leftsite(boundary(tn), tn, site)
-leftsite(::Open, tn::Chain, site::Site) = Site(site.id - 1)
+leftsite(::Union{Open, Periodic}, tn::Chain, site::Site) = Site(site.id - 1)
```

Comment:
Can we canonize a periodic MPS? If so, doing a `::Union{Open,Periodic}` is probably not the best solution.

Reply:
We can, but we need to include the proper logic on the `canonize` function. Nevertheless, we can always do `leftsite` and `rightsite` of  `Periodic` `MPS`.

However, maybe it would be a good idea to put here a check so we don't go to the left of the `Site(1)`, in the case of `Open` `MPS`.

Reply:
I think it would be better to do 2 separa methods:
- One, the `leftsite(::Open)` as it is now
- The other `leftsite(::Periodic)` that wraps the `i` with the period

Reply:
Done, what do you think?

Reply:
Much better, but we should check whether `site` is in range.

Reply:
This is not necessary, right? In `Periodic` we can assume `Site(4)` is effectively `Site(1)`, for a periodic cell of `3` sites.

Reply:
We can do that, but we have to stick to it then.